baseplate/lib/experiments/providers/simple_experiment.py

```diff
@@ -168,14 +178,25 @@ def __init__(
 
         self.variant_set = variant_set
 
-        self._seed = bucket_seed
-        if self._seed is None:
+        if bucket_seed:
+            self._seed = bucket_seed
+        else:
             self._seed = f"{id}.{name}.{shuffle_version}"
 
     @classmethod
-    def from_dict(cls, id, name, owner, start_ts, stop_ts, config, variant_type, enabled=True):
+    def from_dict(
+        cls,
+        id: int,
+        name: str,
+        owner: str,
+        start_ts: int,
+        stop_ts: int,
+        config: Dict[str, Any],
+        variant_type: str,
+        enabled: bool = True,
+    ) -> "SimpleExperiment":
         bucket_val = config.get("bucket_val", "user_id")
-        version = config.get("experiment_version")
+        version = config["experiment_version"]
```

Comment:
You'll also see this a few times in experiment code where we get rid of an `Optional` on the type by using the subscript form for a dictionary. These cases should all be pulling out data that's required so it should be the same effect.